### PR TITLE
Feature/2359 device storage full v2

### PR DESCRIPTION
**Pull request by SwissMountainFrog**
## Description
If SQLlite runs out of storrage capacity, this error now is handled and shown to the user as an alert view. (JIRA ticket https://jira.itc.sap.com/browse/EXPOSUREAPP-2359)

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Developer Menu/Features/DMDeltaOnboardingViewController.swift [hunk 1]
```diff
@@ -45,7 +45,7 @@ class DMDeltaOnboardingViewController: UIViewController, UITextFieldDelegate {
 	override func viewDidLoad() {
 		super.viewDidLoad()
 
-		view.backgroundColor = .white
+		view.backgroundColor = .systemBackground
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Developer Menu/Features/DMDeltaOnboardingViewController.swift:48]: 👍

File: src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesSQLLiteStoreV1.swift [hunk 2]
```diff
@@ -155,16 +174,36 @@ extension DownloadedPackagesSQLLiteStoreV1: DownloadedPackagesStoreV1 {
...
+			if let error = SQLiteError(rawValue: errorCode) {
+				completion?(error)
+			} else {
+				completion?(.unknown)
+			}
 		}
+	}
 
+	private func completeAsync(completion: ((SQLiteError?) -> Void)?) {
+		DispatchQueue.global().async {
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesSQLLiteStoreV1.swift:204]: `global` is not the main thread! Did you mean `main`:

```suggestion
		DispatchQueue.main.async {
```

File: src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesStoreV1.swift [hunk 3]
```diff
@@ -18,15 +18,33 @@
 import Foundation
 import FMDB
 
+enum SQLiteError: Int32 {
+	case sqlite_full = 13
+	case unknown = -1
+}
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesStoreV1.swift:24]: Can you please add documentation where to find these values.

Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesStoreV1.swift:24]: Also, is `Int32` needed? Name and type of this enum is ambiguous. What about:

```suggestion
enum SQLiteErrorCode: Int32 {
  	case sqlite_full = 13
  	case unknown = -1
}
```
(please check white spaces as this web-foo adds some)

Comment (whiskey) [reply src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesStoreV1.swift:24]: or…

```suggestion
enum SQLiteError: Error, Int32 {
  	case sqlite_full = 13
  	case unknown = -1
}
```

File: src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/__tests__/DownloadedPackagesSQLLiteStoreTests.swift [hunk 4]
```diff
@@ -49,6 +49,57 @@ final class DownloadedPackagesSQLLiteStoreTests: XCTestCase {
...
+
+		let package = SAPDownloadedPackage(
+			keysBin: keysBin,
+			signature: signature
+		)
+		
+		let completionExpectation = expectation(description: "Completion callback")
+		let mockStore = MockTestStore()
+		
+		mockStore.fakeSQLiteError = 13
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/__tests__/DownloadedPackagesSQLLiteStoreTests.swift:92]: Magic number!?

```suggestion
		mockStore.fakeSQLiteError = SQLiteError. sqlite_full.rawValue
```

**Review by whiskey — CHANGES_REQUESTED** [review]
I'm a big fan of SQLite and persistent data but this foo causes more issues than solving…

File: src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesSQLLiteStoreV1.swift (hunk 2 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesSQLLiteStoreV1.swift:204]: Refactored it as discussed

File: src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesStoreV1.swift (hunk 3 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/V1/DownloadedPackagesStoreV1.swift]: 👍 
Renamed the enum name to SQLiteErrorCode

File: src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/__tests__/DownloadedPackagesSQLLiteStoreTests.swift (hunk 4 above)
Comment (SwissMountainFrog) [reply src/xcode/ENA/ENA/Source/Services/DownloadedPackagesStore/__tests__/DownloadedPackagesSQLLiteStoreTests.swift]: 😄  Good hint.

**Review by whiskey — APPROVED** [review]

**Review by ArturFriesen — APPROVED** [review]

**Review by janetback — APPROVED** [review]
UA approved

**Pull request merged**